### PR TITLE
Add blog about Node Log Query alpha

### PR DESCRIPTION
**Pull request by aravindhp**
- Enhancement:  https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/2258-node-service-log-viewer
- Feature PR: https://github.com/kubernetes/kubernetes/pull/96120

### CODE REVIEW

**Comment by aravindhp** [comment]
/retitle Add blog about Node Log Query alpha

**Comment by Nancy-Chauhan** [comment]
Hello, Comms Shadow for the 1.27 release here. This feature blog is tracked for release, the deadline for submitting the draft is 4th of April- the sooner the better since there's still editing to be done afterwards 😃 Any doubt, the Comms team is here to help. CC @harshitasao

**Comment by harshitasao** [comment]
Hello @aravindhp, we're doing a global reminder about submitting a draft for review for all opted-in feature blogs. If it's at all possible, it is very helpful for the release team to have drafts submitted for review before the hard deadline date, to better plan the release dates and avoid missing out. Thank you!

**Comment by aravindhp** [comment]
@harshitasao I will add a draft shortly

**Comment by aravindhp** [comment]
@Nancy-Chauhan @harshitasao I have pushed the blog. Please review.

File: content/en/blog/_posts/2023-03-06-node-log-viewer-alpha.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+kubectl get --raw "/api/v1/nodes/node-1.example/proxy/logs/?query=kubelet&pattern=error"
+```
+
+You can also fetch files from `/var/log/` on a Linux node:
+```shell
+kubectl get --raw "/api/v1/nodes/<insert-node-name-here>/proxy/logs/?query=/<insert-log-file-name-here>"
+```
+
+You can read the
+[documentation](content/en/docs/concepts/cluster-administration/system-logs.md#log-query)
```
Comment (sftim) [inline content/en/blog/_posts/2023-03-06-node-log-viewer-alpha.md:65]:
```suggestion
[documentation](/en/docs/concepts/cluster-administration/system-logs/#log-query)
```

File: content/en/blog/_posts/2023-03-06-node-log-viewer-alpha.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+---
+layout: blog
+title: "Node Log Viewer"
```
Comment (sftim) [inline content/en/blog/_posts/2023-03-06-node-log-viewer-alpha.md:3]:
```suggestion
title: "Kubernetes 1.27: Access Node Logs Using The Kubernetes API"
```
“viewer” implies there's a client tool available.

File: content/en/blog/_posts/2023-03-06-node-log-viewer-alpha.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+filtering the logs. The kubelet also uses heuristics to retrieve the logs.
+If the user is not aware if a given system services logs to a file or to the
+native system logger, the heuristics first checks the native operating system
+logger and if that is not available it attempts to retrieve the first logs
+from `/var/log/<servicename>` or `/var/log/<servicename>.log` or
+`/var/log/<servicename>/<servicename>.log`.
+
+On Linux we assume that service logs are available via journald. On Windows we
+assume that service logs are available in the application log provider. Note
+that this feature requires you to have access to node objects.
```
Comment (sftim) [inline content/en/blog/_posts/2023-03-06-node-log-viewer-alpha.md:38]:
```suggestion
On Linux we assume that service logs are available via journald, and that
`journalctl` is installed. On Windows we assume that service logs are available in the
application log provider. Also note that fetching node logs is only available if you are authorized
to do so (in RBAC, that's **get** and **create** access to `nodes/proxy`). The privileges that you
need to fetch node logs also allow elevation-of-privilege attacks, so be careful about how you 
manage them.
```

**Review by sftim — COMMENTED** [review]
A few thoughts.

**Comment by sftim** [comment]
/hold
pending assignment of publication date

File: content/en/blog/_posts/2023-03-06-node-log-viewer-alpha.md (hunk 2 above)
Comment (aravindhp) [reply content/en/blog/_posts/2023-03-06-node-log-viewer-alpha.md]: I renamed this to `Kubernetes 1.27: Query Node Logs Using The Kubelet API` as this is a not a first class Kubernetes API that goes through the API server but is just wired directly through the `proxy/log` endpoint in the kubelet.

**Review by nate-double-u — COMMENTED** [review]
Thanks for making the updates @aravindhp. Reapplying lgtm

/lgtm

File: content/en/blog/_posts/2023-03-06-node-log-viewer-alpha.md (hunk 3 above)
Comment (raesene) [reply content/en/blog/_posts/2023-03-06-node-log-viewer-alpha.md]: I'd agree that cluster operators will need to be very careful if the permission required is create and get on node/proxy. Those rights allow users to directly contact the kubelet API (bypassing auditing and admission control) and execute commands in any pod on a node (amongst other things)

**Comment by harshitasao** [comment]
Hello, Comms Lead for v1.27 here. The publication order and date for the Feature Blog series has been [finalized](https://github.com/kubernetes/sig-release/discussions/2208) and the [tracking board](https://github.com/orgs/kubernetes/projects/117/views/4?filterQuery=status%3ATracked+blog-status%3A%22Ready+for+Review%22) is updated.

The publication date for this article is 21-04-2023 (April 21).

Thank you!

File: content/en/blog/_posts/2023-03-06-node-log-query-alpha.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
+---
+layout: blog
+title: "Kubernetes 1.27: Query Node Logs Using The Kubelet API"
+date: 2023-06-03
```
Comment (harshitasao) [inline content/en/blog/_posts/2023-03-06-node-log-query-alpha.md:4]:
```suggestion
date: 2023-04-21
```

Comment (sftim) [reply content/en/blog/_posts/2023-03-06-node-log-query-alpha.md:4]: Please also change the path (to the Markdown file) to match this date.

File: content/en/blog/_posts/2023-03-06-node-log-query-alpha.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+kubectl get --raw "/api/v1/nodes/node-1.example/proxy/logs/?query=kubelet&pattern=error"
+```
+
+You can also fetch files from `/var/log/` on a Linux node:
+```shell
+kubectl get --raw "/api/v1/nodes/<insert-node-name-here>/proxy/logs/?query=/<insert-log-file-name-here>"
+```
+
+You can read the
+[documentation](/en/docs/concepts/cluster-administration/system-logs/#log-query)
```
Comment (sftim) [inline content/en/blog/_posts/2023-03-06-node-log-query-alpha.md:69]:
```suggestion
[documentation](/docs/concepts/cluster-administration/system-logs/#log-query)
```

**Review by sftim — COMMENTED** [review]
One nit.

File: content/en/blog/_posts/2023-03-06-node-log-query-alpha.md (hunk 4 above)
Comment (aravindhp) [reply content/en/blog/_posts/2023-03-06-node-log-query-alpha.md]: Done

File: content/en/blog/_posts/2023-03-06-node-log-query-alpha.md (hunk 5 above)
Comment (aravindhp) [reply content/en/blog/_posts/2023-03-06-node-log-query-alpha.md]: Fixed

File: content/en/blog/_posts/2023-03-06-node-log-viewer-alpha.md (hunk 3 above)
Comment (aravindhp) [reply content/en/blog/_posts/2023-03-06-node-log-viewer-alpha.md]: I want to call out that the ability read logs using the `/proxy/logs` endpoint is already available and enabled by default. This feature is extending it be able to read journald logs and defaults to false.

**Comment by sftim** [comment]
Thanks
/lgtm

**Pull request merged**